sklearn/utils/_testing.py

```diff
@@ -866,3 +872,146 @@ def __exit__(self, exc_type, exc_value, _):
             self.raised_and_matched = True
 
         return True
+
+
+class MinimalClassifier:
+    """Minimal classifier implementation with inheriting from BaseEstimator.
+
+    This estimator should be tested with:
+
+    * `check_estimator` in `test_estimator_checks.py`;
+    * within a `Pipeline` in `test_pipeline.py`;
+    * within a `SearchCV` in `test_search.py`.
+    """
+    _estimator_type = "classifier"
+
+    def __init__(self, param=None):
+        self.param = param
+
+    def __repr__(self):
+        # Only required when using pytest-xdist to get an id not associated
+        # with the memory location
+        return self.__class__.__name__
```

Comment:
codecov says it's not needed anymore. Let's remove this to keep the code simpler. We can always readd this later if needed.